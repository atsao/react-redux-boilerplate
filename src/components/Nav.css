@import '_sample';

.nav {
  margin-top: 8px;
}

.nav a {
  margin-right: 8px;
}
