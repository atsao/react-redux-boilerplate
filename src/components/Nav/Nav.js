import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

import styles from './Nav.css';

export class Nav extends Component {
  render() {
    return (
      <nav className={styles.nav}>
        <NavLink
          exact
          to="/"
          className={`${styles['btn']} ${styles['btn-ghost']}`}
          activeClassName={styles['selected']}
        >
          Home
        </NavLink>
        <NavLink
          to="/posts"
          className={`${styles['btn']} ${styles['btn-ghost']}`}
          activeClassName={styles['selected']}
        >
          Posts
        </NavLink>
      </nav>
    );
  }
}

export default Nav;
