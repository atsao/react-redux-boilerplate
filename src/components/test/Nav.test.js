import React from 'react';
import { shallow } from 'enzyme';

import Nav from '../Nav';

describe('Nav', () => {
  it('renders without crashing', () => {
    const tree = shallow(<Nav />);
    expect(tree).toMatchSnapshot();
  });
});
