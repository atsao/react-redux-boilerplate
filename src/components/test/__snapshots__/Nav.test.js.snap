// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <nav
    className="nav"
>
    <NavLink
        activeClassName="selected"
        className="btn btn-ghost"
        exact={true}
        to="/"
    >
        Home
    </NavLink>
    <NavLink
        activeClassName="selected"
        className="btn btn-ghost"
        to="/posts"
    >
        Posts
    </NavLink>
</nav>,
  "nodes": Array [
    <nav
      className="nav"
>
      <NavLink
            activeClassName="selected"
            className="btn btn-ghost"
            exact={true}
            to="/"
      >
            Home
      </NavLink>
      <NavLink
            activeClassName="selected"
            className="btn btn-ghost"
            to="/posts"
      >
            Posts
      </NavLink>
</nav>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Nav />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Nav {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <nav
          className="nav"
>
          <NavLink
                    activeClassName="selected"
                    className="btn btn-ghost"
                    exact={true}
                    to="/"
          >
                    Home
          </NavLink>
          <NavLink
                    activeClassName="selected"
                    className="btn btn-ghost"
                    to="/posts"
          >
                    Posts
          </NavLink>
</nav>,
        "_debugID": 2,
        "_renderedOutput": <nav
          className="nav"
>
          <NavLink
                    activeClassName="selected"
                    className="btn btn-ghost"
                    exact={true}
                    to="/"
          >
                    Home
          </NavLink>
          <NavLink
                    activeClassName="selected"
                    className="btn btn-ghost"
                    to="/posts"
          >
                    Posts
          </NavLink>
</nav>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Nav />,
}
`;
